Source/Core/loadKTX.js

```diff
@@ -0,0 +1,198 @@
+        var byteBuffer = new Uint8Array(data);
+
+        var isKTX = true;
+        for (var i = 0; i < fileIdentifier.length; ++i) {
+            if (fileIdentifier[i] !== byteBuffer[i]) {
+                isKTX = false;
+                break;
+            }
+        }
+
+        if (!isKTX) {
+            throw new RuntimeError('Invalid KTX file.');
+        }
+
+        var view;
+        var byteOffset;
+
+        if (defined(data.buffer)) {
+            view = new DataView(data.buffer);
+            byteOffset = data.byteOffset;
+        } else {
+            view = new DataView(data);
+            byteOffset = 0;
+        }
+
+        byteOffset += 12; // skip identifier
+
+        var endianness = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        if (endianness !== endiannessTest) {
+            // TODO: Switch endianness?
+            throw new RuntimeError('File is the wrong endianness.');
+        }
+
+        var glType = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var glTypeSize = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var glFormat = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var glInternalFormat = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var glBaseInternalFormat = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var pixelWidth = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var pixelHeight = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var pixelDepth = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var numberOfArrayElements = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var numberOfFaces = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var numberOfMipmapLevels = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var bytesOfKeyValueByteSize = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        // TODO: read metadata? At least need to check for KTXorientation
+        byteOffset += bytesOfKeyValueByteSize;
+
+        var imageSize = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var texture = new Uint8Array(data, byteOffset, imageSize);
+
+        // Some tools use a sized internal format.
+        // See table 2: https://www.opengl.org/sdk/docs/man/html/glTexImage2D.xhtml
+        if (glInternalFormat === 0x8051) {         // GL_RGB8
+            glInternalFormat = PixelFormat.RGB;
+        } else if (glInternalFormat === 0x8058) {  // GL_RGBA8
+            glInternalFormat = PixelFormat.RGBA;
+        }
+
+        if (!PixelFormat.validate(glInternalFormat)) {
+            throw new RuntimeError('glInternalFormat is not a valid format.');
```

Comment:
Wouldn't all file loading/parsing be a runtime error? The developer might not have control over the file. KML and GeoJSON work this way. Shouldn't parsing a 3D tile also throw  runtime errors when they fail to parse?